README.md

```diff
@@ -23,7 +23,7 @@ Basic connections, and those with options. The compiler will force these to be c
 ```rust
 let nc = nats::connect("localhost")?;
 
-let nc2 = nats::Connection::new()
+let nc2 = nats::Options::new()
```

Comment:
We originally had options in the Go client and then called connect on those, we wen away from that so don't want to introduce that here.

I chose the type state pattern to allow different states of a connection for compile time. For reconnect logic this will be runtime state (I have an enum) that will apply to a connected connection.